test-apps/imodel-transformer/src/Main.ts

```diff
@@ -139,33 +160,33 @@ void (async () => {
       if (args.sourceStartChangesetIndex || args.sourceStartChangesetId) {
         assert(!(args.sourceStartChangesetIndex && args.sourceStartChangesetId), "Pick single way to specify starting changeset");
         if (args.sourceStartChangesetIndex) {
-          args.sourceStartChangesetId = await IModelHubUtils.queryChangesetId(accessToken, sourceIModelId, args.sourceStartChangesetIndex);
+          args.sourceStartChangesetId = await IModelHubUtils.queryChangesetId(accessToken!, sourceIModelId, args.sourceStartChangesetIndex);
```

Comment:
Nit - prefer assert to bang

Reply:
as in prefer `(accessToken as AccessToken)`? I'm personally in the bang camp. I think if we prefer in this repo explicit casts over bangs/nonnull-assertions, we should use the eslint hook that disallows it which we I don't think we are.

Reply:
I just confirmed it's set to off in `tools/eslint-plugin/dist/configs/itwinjs-recommended.js`. Please resolve if you think it's ok to keep the non-null assertions

Reply:
`assert(undefined !== accessToken);`
We are gradually rolling out no-non-null assertion (aka anti-bang) lint rule.

Reply:
Although really shouldn't you bailing with an error if you fail to obtain an access token when one is required?

Reply:
out of curiosity, what's the reason? I've always preferred non-null assertions when using the rule that disallows it. I'll go remove the new ones from this PR

Reply:
Far too often we have had to track down and post emergency fixes for uncaught exceptions in production caused by folks slapping a "bang" in to shut the compiler up when it's warning them about a real potential problem in their code.
Compile-time errors >>>> run-time errors.
Exclamation marks are far too easy to overlook during code review.

Reply:
It's a shame because I tend to dislike the highly verbose casts you end up with,
but definitely an exclamation point is easy to miss in the diff, so I agree with the change.
Thanks for the explanation.

Reply:
although in this case AccessToken should not be allowed to be undefined. See my change above.

Reply:
one moment, I am currently in process of rewriting it to not need any assertions at all, by lazily async fetching the accessToken. I will push the update in a moment, then check your comment since I'm almost done

Reply:
I pushed the changes, please take a look. The issue came from typescript not realizing that
`args.sourceITwinId` being true implies that `requiresLogin` is true which implies that `accessToken` is set and not undefined.

I removed the need for requiresLogin by putting access to accessToken behind a promise that is always going to lazily login when asked for. Now accessToken doesn't need to be undefined ever since if it isn't ready it can be fetched and is guaranteed to be valid or reject.

As for the empty string being an invalid access token, I was unaware of that. I think in this case it is preferable to never store an invalid state of the accessToken since with lazy fetching I don't have to.

Reply:
@mgooding can I resolve this since I no longer use the non-null assertions? discussion about getting the access token will be followed up in the other thread